scrapy/contrib/periodicstats/collectors.py

```diff
@@ -0,0 +1,152 @@
+from collections import defaultdict
+import re
+
+from twisted.internet import task
+
+from scrapy.statscol import MemoryStatsCollector
+from scrapy.exceptions import NotConfigured
+from scrapy.utils.misc import load_object
+from scrapy.contrib.periodicstats.observers import PeriodicStatsObserver
+from scrapy.contrib.periodicstats.pipelines import PeriodicStatsPipeline
+
+
+DEFAULT_PERIODIC_STATS_ENABLED = True
+DEFAULT_PERIODIC_STATS_EXPORT_ALL = False
+DEFAULT_PERIODIC_STATS_EXPORT_ALL_INTERVAL = 60
+
+
+class PeriodicStatsCollector(MemoryStatsCollector):
+    """
+    Stats collector class that periodically generates stats and pass them to pipelines for processing
+    """
+
+    def __init__(self, crawler):
+        super(PeriodicStatsCollector, self).__init__(crawler=crawler)
+        self._enabled = crawler.settings.get('PERIODIC_STATS_ENABLED',
+                                             DEFAULT_PERIODIC_STATS_ENABLED)
+        if self._enabled:
+            self._export_all = crawler.settings.get('PERIODIC_STATS_EXPORT_ALL',
+                                                    DEFAULT_PERIODIC_STATS_EXPORT_ALL)
+            self._export_all_interval = crawler.settings.get('PERIODIC_STATS_EXPORT_ALL_INTERVAL',
+                                                             DEFAULT_PERIODIC_STATS_EXPORT_ALL_INTERVAL)
+            self._load_observers(crawler)
+            self._load_pipelines(crawler)
+            self._interval = 0
+
+    def open_spider(self, spider):
+        super(PeriodicStatsCollector, self).open_spider(spider)
+        if self._enabled:
+            for pipeline in self._pipelines:
+                pipeline.open_spider(spider)
+            self._task = task.LoopingCall(self._process_interval_stats, spider)
+            self._task.start(1)
```

Comment:
`LoopingCall.start()` returns a deferred that will be fired in case a (temporal) error occurs with the called function, you have to trap the error, log it and re-setup the looping call or it will stop updating stats for the rest of the crawling.